src/main/java/com/exasol/adapter/dynamodb/documentfetcher/dynamodb/DynamodbQueryDocumentFetcher.java

```diff
@@ -57,12 +57,14 @@ private void addKeyConditionAndFilterExpression(final RemoteTableQuery<DynamodbN
             final DynamodbIndex mostRestrictedIndex) {
         final QueryPredicate<DynamodbNodeVisitor> selectionOnIndex = new DynamodbQuerySelectionFilter()
                 .filter(documentQuery.getSelection(), getIndexPropertyNameWhitelist(mostRestrictedIndex));
-        final DynamodbValueListBuilder valueListBuilder = new DynamodbValueListBuilder();
+        final DynamodbAttributeNamePlaceholderMapBuilder namePlaceholderBuilder = new DynamodbAttributeNamePlaceholderMapBuilder();
+        final DynamodbAttributeValuePlaceholderMapBuilder valuePlaceholderBuilder = new DynamodbAttributeValuePlaceholderMapBuilder();
         final DynamodbFilterExpressionFactory filterExpressionFactory = new DynamodbFilterExpressionFactory();
         final String keyConditionExpression = filterExpressionFactory.buildFilterExpression(selectionOnIndex,
-                valueListBuilder);
+                namePlaceholderBuilder, valuePlaceholderBuilder);
         this.queryRequest.setKeyConditionExpression(keyConditionExpression);
-        this.queryRequest.setExpressionAttributeValues(valueListBuilder.getValueMap());
+        this.queryRequest.setExpressionAttributeNames(namePlaceholderBuilder.getValueMap());
+        this.queryRequest.setExpressionAttributeValues(valuePlaceholderBuilder.getValueMap());
```

Comment:
Maybe it's time to use a builder instead of setters

Reply:
It's a DynamoDB API object and there is no builder. So I can't change it :-(